### PR TITLE
cranelift: Upgrade libm to 0.2.4

### PR DESCRIPTION
**Pull request by afonso360**
This resolves an issue with incorrect `fmaf` on the x86_64-pc-windows-gnu target under some inputs. 

This was fixed in https://github.com/rust-lang/libm/pull/267

See: #4517

### CODE REVIEW

**Comment by afonso360** [comment]
It looks like this fails (expectedly) in cargo-vet.

However, it is my understanding that only Bytecode Alliance members can modify this / approve new deps. I think this was mentioned in one of the cranelift meetings, but can't find the exact policy in the repository.

@alexcrichton is this the case?

File: cranelift/filetests/filetests/runtests/fma.clif [hunk 1]
```diff
@@ -1,6 +1,8 @@
+test interpret
 test run
 target aarch64
 target s390x
+target x86_64 has_avx=false has_fma=false
```
Comment (bjorn3) [inline cranelift/filetests/filetests/runtests/fma.clif:5]: Can you also check it with fma instructions?

Comment (afonso360) [reply cranelift/filetests/filetests/runtests/fma.clif:5]: Yes! As soon as https://github.com/bytecodealliance/wasmtime/pull/4539 is merged.

Although that needs a rebase.

**Review by alexcrichton — APPROVED** [review]
Thanks!

If you add this diff to this PR that should appease `cargo vet`:

```diff
diff --git a/supply-chain/audits.toml b/supply-chain/audits.toml
index 9a4b9bd23..c172585c1 100644
--- a/supply-chain/audits.toml
+++ b/supply-chain/audits.toml
@@ -19,6 +19,16 @@ criteria = "safe-to-deploy"
 version = "1.0.0"
 notes = "I am the author of this crate."

+[[audits.libm]]
+who = "Alex Crichton <alex@alexcrichton.com>"
+criteria = "safe-to-deploy"
+delta = "0.2.2 -> 0.2.4"
+notes = """
+This diff primarily fixes a few issues with the `fma`-related functions,
+but also contains some other minor fixes as well. Everything looks A-OK and
+as expected.
+"""
+
 [[audits.regalloc2]]
 who = "Jamey Sharp <jsharp@fastly.com>"
 criteria = "safe-to-deploy"
```

**Pull request merged**